fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/ConvertJoinByCondition.java

```diff
@@ -26,26 +26,28 @@
 import java.util.List;
 
 /**
- * Convert invalid inner join to cross join.
+ * Convert join to cross join.
  * Like: A inner join B on true -> A cross join B on true;
+ * Like: A left join B on true -> A cross join B on true;
  * Or
  * Convert invalid cross join to inner join.
  * Like: A cross join B on A.id > 1 -> A inner join B on A.id > 1;
  */
-public class ConvertInnerOrCrossJoin implements RewriteRuleFactory {
+public class ConvertJoinByCondition implements RewriteRuleFactory {
```

Comment:
outer join cannot convert to cross join even condition is true. consider left outer join with empty right child. the result is different.